### PR TITLE
Add toggle theme function

### PR DESCRIPTION
**Pull request by zi-yii**
Our app supports only a dark theme, which limits personalisation for users.

Let's introduce a theme toggle functionality that allows users to switch between light and dark themes according to their preference.

With this update, users can easily select their preferred theme, and their choice will be saved in their user preferences. The selected theme will automatically load the next time the app is started to improve user experience.

This enhancement increases user satisfaction by allowing them to customise the appearance of the app to their liking.

### CODE REVIEW

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 1]
```diff
@@ -24,12 +27,20 @@
...
+    private static final String LIGHT_THEME_PATH = "/view/LightTheme.css";
 
     private final Logger logger = LogsCenter.getLogger(getClass());
 
     private Stage primaryStage;
     private Logic logic;
 
+    //private String theme = "light";
+    private String theme;
+    private Scene scene;
```
Comment (RezwanAhmed123) [inline src/main/java/seedu/address/ui/MainWindow.java:42]: Is there any significance behind this comment? Apologies, but I am having trouble understanding this.

File: src/main/java/seedu/address/ui/MainWindow.java [hunk 2]
```diff
@@ -167,6 +186,64 @@ public ClientListPanel getClientListPanel() {
...
+            scene.getStylesheets().add(getClass().getResource(DARK_THEME_PATH).toExternalForm());
+            theme = DARK_THEME;
+
+            // Get user prefs, set the theme, and update
+            UserPrefs updatedPrefs = (UserPrefs) logic.getUserPrefs();
+            logic.setUserPrefs(updatedPrefs.setTheme(DARK_THEME));
+
+            resultDisplay.setFeedbackToUser("Theme changed to " + theme + " mode.");
+        } else {
+            resultDisplay.setFeedbackToUser("Theme is already set to " + theme + " mode.");
```
Comment (RezwanAhmed123) [inline src/main/java/seedu/address/ui/MainWindow.java:243]: Is there a way to either abstract out the similar code, or handling this as a toggle? It seems like a lot of lines are being repeated, which may make maintenance difficult later. What do you think?

**Review by RezwanAhmed123 — CHANGES_REQUESTED** [review]
LGTM, some minor code quality changes.

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 1 above)
Comment (zi-yii) [reply src/main/java/seedu/address/ui/MainWindow.java:42]: oops, missed out on removing this

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 2 above)
Comment (yanqiyqh) [reply src/main/java/seedu/address/ui/MainWindow.java:243]: I think having `public static final String THEME_CHANGE_MESSAGE = "Theme changed to %1$s mode.` will be a good way to abstract it out, as well as keeping our String formatting in line with our other messages

**Review by yanqiyqh — COMMENTED** [review]
Great work!

File: src/main/java/seedu/address/ui/MainWindow.java (hunk 2 above)
Comment (zi-yii) [reply src/main/java/seedu/address/ui/MainWindow.java:243]: Great suggestion, will make use of that!

**Review by yanqiyqh — APPROVED** [review]
LGTM 👍

**Review by RezwanAhmed123 — APPROVED** [review]
LGTM!

**Pull request merged**